### PR TITLE
*: introduce retry logic and simplify providers

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

**Comment by vcaputo** [comment]
Looks reasonable to me, I never liked the provider stuff, it was super verbose repetitive scaffolding that didn't add anything AFAICT.

**Comment by ajeddeloh** [comment]
The vmware_unsupported provider needs to be updated.

File: internal/util/http.go [hunk 1]
```diff
@@ -26,6 +27,14 @@ import (
 	"github.com/coreos/ignition/internal/version"
 )
 
+const (
+	maxAttempts = 15
+)
```
Comment (ajeddeloh) [inline internal/util/http.go]: Should this be a property of a client in case we want to make it configurable later?

**Comment by crawford** [comment]
This should be ready to go. Please pay special attention to c730b50e9421c806a5039311286da7f0c9583e7f. It will retry a request for roughly one minute before giving up.

File: internal/util/http.go (hunk 1 above)
Comment (crawford) [inline internal/util/http.go]: Probably. Let's change it when we do the configurable timeouts.

**Comment by ajeddeloh** [comment]
Only thing I can think of is that the `c.client.Do` will block until headers have been received or the request times out if the server accepts the connection but doesnt actually send data. This means the total time is going to be longer than expected. I don't think its a big deal or even a problem, we should just know that the max time is not `(n_retries * avg_delay)` its `(n_retries * (avg_delay + header_timeout))`.  Otherwise LGTM.

**Pull request merged**